security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -162,3 +150,44 @@ func (c *citadelClient) getTLSDialOption() (grpc.DialOption, error) {
 	transportCreds := credentials.NewTLS(&config)
 	return grpc.WithTransportCredentials(transportCreds), nil
 }
+
+func (c *citadelClient) Reconnect() error {
+	err := c.releaseResource()
+	if err != nil {
+		return fmt.Errorf("failed to close connection")
+	}
+
+	conn, err := c.buildConnection()
+	if err != nil {
+		return err
+	}
+	c.conn = conn
+	c.client = pb.NewIstioCertificateServiceClient(conn)
+	return err
+}
+
+func (c *citadelClient) releaseResource() error {
```

Comment:
This is just a one-line function and only used in Reconnect(), doesn't seem any useful, can you embed the logic just there?